hazelcast/src/main/java/com/hazelcast/version/IllegalVersionException.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.version;
+
+import com.hazelcast.core.HazelcastException;
+
+/**
+ * Exception thrown whenever there is a Version or ClusterVersion mismatch
+ */
+public class IllegalVersionException extends HazelcastException {
+
+    public IllegalVersionException() {
+    }
+
+    public IllegalVersionException(final String message) {
+        super(message);
+    }
+
+    public IllegalVersionException(final String message, final Throwable cause) {
```

Comment:
I didn't check everything, but unused constructors will hurt the code coverage. I also think Peter has the rule of thumb to not have them, regardless of the code coverage. So in the end you might want to remove the unused ones.